### ISSUE TITLE
Cannot start container: [8] System error: no such file or directory

### ISSUE THREAD

**Issue by koehn** [docker/docker#18098]
Description of problem:
When trying to run my project (clone from [here](https://github.com/koehn/diaspora-docker.git)), I get a `Cannot start container [id]: [8] System error: no such file or directory`

This occurs on Mac OS X and on Ubuntu 14.04, when starting the container with `docker-compose run  --entrypoint /bin/bash diaspora` or any other entrypoint (or none at all). 

`docker version`:

Mac OS X:

    Client:
     Version:      1.9.0
     API version:  1.21
     Go version:   go1.4.2
     Git commit:   76d6bc9
     Built:        Tue Nov  3 17:43:42 UTC 2015
     OS/Arch:      linux/amd64

    Server:
     Version:      1.9.0
     API version:  1.21
     Go version:   go1.4.2
     Git commit:   76d6bc9
     Built:        Tue Nov  3 17:43:42 UTC 2015
     OS/Arch:      linux/amd64
     
Ubuntu 14.04: 

    Client:
     Version:      1.9.0
     API version:  1.21
     Go version:   go1.4.3
     Git commit:   76d6bc9
     Built:        Tue Nov  3 19:20:09 UTC 2015
     OS/Arch:      darwin/amd64

    Server:
     Version:      1.9.0
     API version:  1.21
     Go version:   go1.4.3
     Git commit:   76d6bc9
     Built:        Tue Nov  3 19:20:09 UTC 2015
     OS/Arch:      linux/amd64

`docker info`:
Mac OS X:

    Containers: 6
    Images: 59
    Server Version: 1.9.0
    Storage Driver: aufs
     Root Dir: /mnt/sda1/var/lib/docker/aufs
     Backing Filesystem: extfs
     Dirs: 71
     Dirperm1 Supported: true
    Execution Driver: native-0.2
    Logging Driver: json-file
    Kernel Version: 4.1.12-boot2docker
    Operating System: Boot2Docker 1.9.0 (TCL 6.4); master : 16e4a2a - Tue Nov  3 19:49:22 UTC 2015
    CPUs: 1
    Total Memory: 1.956 GiB
    Name: default
    ID: GBER:G2VE:OF4T:7MHR:67DD:WX2O:FBX5:SQ5V:5QSS:WBRX:JMFR:DZVE
    Debug mode (server): true
     File Descriptors: 24
     Goroutines: 48
     System Time: 2015-11-19T18:59:03.549401764Z
     EventsListeners: 0
     Init SHA1: 
     Init Path: /usr/local/bin/docker
     Docker Root Dir: /mnt/sda1/var/lib/docker
    Labels:
     provider=virtualbox

Ubuntu 14.04:

     Containers: 13
     Images: 50
     Server Version: 1.9.0
     Storage Driver: aufs
      Root Dir: /var/lib/docker/aufs
      Backing Filesystem: extfs
      Dirs: 78
      Dirperm1 Supported: false
     Execution Driver: native-0.2
     Logging Driver: json-file
     Kernel Version: 3.13.0-66-generic
     Operating System: Ubuntu 14.04.3 LTS
     CPUs: 4
     Total Memory: 15.43 GiB
     Name: dq.koehn.com
     ID: GSWJ:7YFW:ITN7:O4O6:XZGO:4XYL:5DMN:RUNQ:5VN3:4W3R:BBYX:4FPL
     WARNING: No swap limit support

`uname -a`:

    Darwin Fungo.local 15.0.0 Darwin Kernel Version 15.0.0: Sat Sep 19 15:53:46 PDT 2015; root:xnu-3247.10.11~1/RELEASE_X86_64 x86_64

    Linux foo 3.13.0-66-generic #108-Ubuntu SMP Wed Oct 7 15:20:27 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Environment details (AWS, VirtualBox, physical, etc.):
Physical hardware in both cases.

How reproducible:
Every time, regardless of entrypoint. 

Steps to Reproduce:
1. `git clone https://github.com/koehn/diaspora-docker.git`
2. `cd diaspora-docker/docker`
3. `docker-compose run  --entrypoint /bin/bash diaspora`

Actual Results:
`ERROR: compose.cli.main.main: Cannot start container foo: [8] System error: no such file or directory`

Expected Results:
Running container, or some reasonable error message. 

Additional info:

**Comment by bitglue** [comment]
I ran into something very similar. It looks like one of the image layers referenced by the image I was trying to run does not actually exist. It was resolved by deleting that image (`docker rmi ...`) and then pulling it again.

This busted state seems to be precipitated by running concurrent `docker pull` processes. Image pulling is so incredibly buggy that I've resorted to serializing all of my pulls with `flock`.

Running 1.7.1 from CoreOS stable here.

**Comment by koehn** [comment]
I removed all containers and images and tried again. Same exact result.

**Comment by superseb** [comment]
Which docker-compose version @koehn ?

**Comment by koehn** [comment]
Ubuntu 14.04: `docker-compose version: 1.5.1`
Mac OS X El Capitan: `docker-compose version: 1.5.0`

**Comment by crosbymichael** [comment]
I can reproduce with compose but not with just doing a docker run.  Could be an api thing.

**Comment by crosbymichael** [comment]
#18066 looks related to this issue.

ping @calavera @runcom

**Comment by Sajonara** [comment]
I have the same problem with docker compose 1.5.1 and docker 1.9.1, on OS X with a pulled Ruby2.2.3 instance. When I try to do a docker-compose rake:db migrate it exits with an error and says it cannot start the container. The first rake db:create was done but any migration I would like to add results in this particular error.

As this are my very first steps to Docker this is really frustrating. I have searched Google and Github for a while now and read tons of issues, bug reports, etc. but did not find a solution yet.

Kind regards

**Comment by matthughes** [comment]
Note sure if this is related, but I got the same error when trying to volume mount a leaf directory that didn't exist with docker-compose.  Same operation works fine with docker run

```
# Create the parent directory /tmp/foo
> mkdir -p /tmp/foo

# Try to bind to /tmp/foo/bar on the host.  /tmp/foo exists, just haven't created bar
# with `docker run` it will autocreate this leaf directory
> cat docker-compose.yml
busybox:
  image: busybox
  volumes:
    - /tmp/foo/bar:/foo:Z


> docker-compose up
docker-compose up
Pulling busybox (busybox:latest)...
latest: Pulling from library/busybox
5c5fb281b01e: Pull complete
fc0db02f3072: Pull complete
library/busybox:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
Digest: sha256:e4f93f6ed15a0cdd342f5aae387886fba0ab98af0a102da6276eaf24d6e6ade0
Status: Downloaded newer image for docker.io/busybox:latest

Creating busybox_1
ERROR: no such file or directory

# Yet on the same machine, you can bind volumes of uncreated directories with `docker run`
> docker run -i -t -v /tmp/foo/bar:/foo/bar busybox ls -lta /foo/bar
total 4
drwxr-xr-x    3 root     root          4096 Dec 17 17:13 ..
drwxr-xr-x    2 root     root             6 Dec 17 17:10 .
```

**Comment by abevoelker** [comment]
I seem to be encountering this, but not sure if it's the exact same issue since it's with a straight `docker run` and not `docker-compose`.

I had been running a container I built locally and tagged `foo` (ran with `--rm` each time). I had ran it several times without issue, then after running it once with `--privileged` added (was trying a few things to get `lsof` working inside the container) and exiting that container, my next run (without `--privileged`) encountered this issue. And now I seem to be stuck with it.

![screenshot from 2015-12-31 21 24 15](https://cloud.githubusercontent.com/assets/153459/12069718/f57bf90a-b00c-11e5-8c67-39530bc15679.png)

![screenshot from 2015-12-31 21 32 55](https://cloud.githubusercontent.com/assets/153459/12069723/1fb0ee60-b00d-11e5-881f-aaca8ad6aa4f.png)

![screenshot from 2015-12-31 21 33 07](https://cloud.githubusercontent.com/assets/153459/12069728/300c944e-b00d-11e5-9a0f-b92ee682f8c0.png)

I deleted all images, deleted all containers, did an `apt-get remove --purge docker-engine` as well as manually deleting `/var/lib/docker` and `/etc/docker` (since purge apparently doesn't clean everything up), and even after reinstalling docker I'm still encountering this issue when trying to run any container:

![screenshot from 2015-12-31 21 39 39](https://cloud.githubusercontent.com/assets/153459/12069729/34a07354-b00d-11e5-96d4-9bda0bb32882.png)

![screenshot from 2015-12-31 22 21 05](https://cloud.githubusercontent.com/assets/153459/12069717/dd003904-b00c-11e5-8499-08a099b55d70.png)

My host machine is Ubuntu:

```
$ sudo docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
$ sudo docker info
Containers: 0
Images: 0
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 0
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-33-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 15.29 GiB
Name: abe-home
ID: MABI:HWEI:Q27Z:VQAM:C42P:WFER:GXA3:BEFJ:RINF:5SMF:ID5N:B376
WARNING: No swap limit support
```

**Comment by abevoelker** [comment]
Okay it seems I **can** run containers as long as I pass in `--privileged` to `docker run`. But that doesn't help me when it comes to building images since the build process runs containers unprivileged.

The only other change I had made was unloading the AppArmor profile in `/etc/apparmor.d/docker`, then after that completely unloading all AppArmor profiles (was messing with getting `lsof` to work in containers), but not sure if that is related.

I may actually be running into a separate issue here but with a similar error message. If so, sorry for the noise!

**Comment by abevoelker** [comment]
Okay I figured it out, and it was related to me messing with the docker AppArmor profile and not this issue. Sorry for the noise!

**Comment by EwanValentine** [comment]
I'm getting...

```
Recreating 051010d58e_051010d58e_051010d58e_051010d58e_051010d58e_051010d58e_051010d58e_051010d58e_051010d58e_bandzest_core-api_1
ERROR: Cannot start container 40dcfe9579ccce5fceb7bff3d1e1c33325f6b1b64d0b532e1b0dc9fbf2958274: [8] System error: no such file or directory

```
For no apparent reason. I'm guessing this is related? I'm trying to run a golang app in Docker on Ubuntu 14.04.

**Comment by mbbender** [comment]
I know this is closed but in case it helps anyone, I got this error due to a bad working_dir path in my docker-compose file.

**Comment by robertaistleitner** [comment]
@mbbender your comment helped me out, thanks!

I had a 'similar' issue: I used an incorrect volume mounting point (changed the volume it in the Dockerfile but not in the docker-compose.yml).
Cleaning up the containers + a rebuild fixed it for me.

**Comment by koehn** [comment]
Fixed in 1.10.
